lib/validate_partitions.pm

```diff
@@ -0,0 +1,53 @@
+# SUSE's openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+package validate_partitions;
+use Exporter 'import';
+our @EXPORT = qw(validate_partition_table);
+
+use strict;
+use warnings;
+use scheduler 'get_test_suite_data';
+use testapi;
+use Test::Assert ':all';
+
+
+
+sub validate_partition_table {
+    my ($args) = @_;
+    my $expected_table_type;
+    my $root_disk;
+
+    if (exists($args->{root_disk})) { $root_disk = $args->{root_disk} }
+    else {
+        my @mount_output = (split /\n/, script_output('mount'));
+        my $root_disk;
+        foreach ( @mount_output ) {
+            if ( $_ =~ /(\/dev\/.da).+$/ ) {
+             $root_disk = $1;
+            }
+        }
+    }
+
+    if   (exists($args->{table_type})) { $expected_table_type = $args->{table_type} }
+    else                               { $expected_table_type = (check_var('BACKEND', 's390x')) ? "dasd" : "gpt" }
```

Comment:
same here, even shorter, just `unless ($args->{table_type}) { .. }` or `unless (exists $args->{table_type}) {...}`

Reply:
This thing can be written with ternary completely:

```perl
    $expected_table_type = exists($args->{table_type}) ? $args->{table_type}
      : (check_var('BACKEND', 's390x')) ? "dasd"
      :                                   "gpt";
```

Reply:
I avoid nested ternaries because I find them harder to read than if..elses. Anyway, I'll probably get rid of that part of code (sigh) and create whole lot of yamls, as per discussions.